File: LSP/C#/não aplicado/Retangulo.cs
public class Retangulo
{
	public virtual int Largura { get; set; }
	public virtual int Altura { get; set; }
	public int Area 
	{ 
		get 
		{ 
			return Largura * Altura; 
		} 
	}
}

public class Quadrado : Retangulo
{
	public override int Largura
	{
		get { return base.Largura; }
		set { base.Largura = base.Altura = value; }
	}
	
	public override int Altura
	{
		get { return base.Altura; }
		set { base.Altura = base.Largura = value; }
	}
}

File: LSP/Ruby/não aplicado/retangulo.rb
class Retangulo
    attr_accessor :altura, :largura
        
    def area
        @altura * @largura
    end
end

class Quadrado < Retangulo  
    def altura=(altura)
        @altura = altura
        @largura = altura
    end
    
    def largura=(largura)
        @largura = largura
        @altura = largura
    end
end

